pyvista/plotting/picking.py

```diff
@@ -555,216 +728,892 @@ def enable_surface_picking(
                If enabled, left-clicking will **not** display the bounding box
                around the picked mesh.
 
+        picker : str | PickerType, optional
+            Choice of VTK picker class type:
+
+                * ``'hardware'``: Uses ``vtkHardwarePicker`` which is more
+                  performant for large geometries (default).
+                * ``'cell'``: Uses ``vtkCellPicker``.
+                * ``'point'``: Uses ``vtkPointPicker`` which will snap to
+                  points on the surface of the mesh.
+                * ``'volume'``: Uses ``vtkVolumePicker``.
+
+        use_picker : bool, default: False
+            When ``True``, the callback will also be passed the picker.
+
+        clear_on_no_selection : bool, default: True
+            Clear the selections when no point is selected.
+
         **kwargs : dict, optional
             All remaining keyword arguments are used to control how
             the picked path is interactively displayed.
 
-        Returns
-        -------
-        vtk.vtkPropPicker
-            Property picker.
-
         Notes
         -----
         Picked point can be accessed from :attr:`picked_point
         <pyvista.Plotter.picked_point>` attribute.
 
         Examples
         --------
-        Add a cube to a plot and enable cell picking. Enable ``left_clicking``
-        to immediately start picking on the left click and disable showing the
-        box. You can still press the ``p`` key to select meshes.
+        Add a cube to a plot and enable cell picking.
 
         >>> import pyvista as pv
         >>> cube = pv.Cube()
         >>> pl = pv.Plotter()
         >>> _ = pl.add_mesh(cube)
-        >>> _ = pl.enable_surface_picking(left_clicking=True)
+        >>> _ = pl.enable_surface_point_picking()
 
         See :ref:`surface_picking_example` for a full example using this method.
```

Comment:
Rename this example

```suggestion
        See :ref:`surface_point_picking_example` for a full example using this method.
```

Reply:
The reference of the example is still `surface_picking_example`... either needs to be updated there too or ignored